apps/omg_eth/lib/omg_eth/root_chain/abi.ex

```diff
@@ -29,16 +35,17 @@ defmodule OMG.Eth.RootChain.Abi do
       end)
 
     {function_spec, data} = ABI.find_and_decode(function_specs, enriched_data)
+    decode_function_call_result(function_spec, data)
```

Comment:
this needed a slightly different approach to the calls we're making from Ethereum Aggregator, hence the refactor.
the returned data from events or calls already contains the data that allows us to identify which function selector we should use to decode the data.
for function calls this isn't so, so we expanded decode_function to `decode_function/2`